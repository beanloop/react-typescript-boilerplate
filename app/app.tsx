import * as React from 'react'
import {Router, Route} from 'react-router'

const Container = ({children}) => <main>{children}</main>

export function routes() {
  return (
    <Route path='/' component={Container}>
    </Route>
  )
}

export const App = () => <Router>{routes()}</Router>
