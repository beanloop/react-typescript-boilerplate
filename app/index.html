<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>Culinam</title>
    <style>
      html, body, #app {
        margin: 0;
        width: 100%;
        height: 100%;
      }
    </style>
  </head>
  <body>
    <div id="app"></div>
    <script src="js.js"></script>
  </body>
</html>
