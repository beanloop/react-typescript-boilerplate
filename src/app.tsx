import {BrowserRouter, Link} from 'react-router-dom'
import Route from 'react-router/Route'
import Switch from 'react-router/Switch'

export const Home = () => <div>Home <Link to="/hello">Go to hello</Link></div>
export const Hello = () => <span>Hello, world!</span>

export const Routes = () =>
  <Switch>
    <Route path="/hello" component={Hello} />
    <Route path="/" render={Home} />
  </Switch>

export const App = () =>
  <BrowserRouter>
    <Main></Main>
  </BrowserRouter>


export const Main = () =>
  <main>
    <Routes />
  </main>
