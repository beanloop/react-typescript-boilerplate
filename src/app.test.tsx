import {render} from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import {MemoryRouter} from 'react-router-dom'
import {Main} from './app'

describe('App', () => {
  global['React'] = React

  it('should render Main', () => {
    const wrapper = render(
      <MemoryRouter>
        <Main />
      </MemoryRouter>,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
