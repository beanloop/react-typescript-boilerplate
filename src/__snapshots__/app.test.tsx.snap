// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Main 1`] = `
<main>
  <div>
    Home 
    <a
      href="/hello"
    >
      Go to hello
    </a>
  </div>
</main>
`;
